_studio/shared/umc/codec/av1_dec/src/umc_av1_decoder.cpp

```diff
@@ -0,0 +1,749 @@
+            par.info.color_format = UMC::YUV444;
+        else if (sh.color_config.subsampling_x && !sh.color_config.subsampling_y)
+            par.info.color_format = UMC::YUY2;
+        else if (sh.color_config.subsampling_x && sh.color_config.subsampling_y)
+            par.info.color_format = UMC::NV12;
+
+        if (sh.color_config.BitDepth == 8 && par.info.color_format == UMC::YUV444)
+            par.info.color_format = UMC::AYUV;
+        if (sh.color_config.BitDepth == 10)
+        {
+            switch (par.info.color_format)
+            {
+            case UMC::NV12:   par.info.color_format = UMC::P010; break;
+            case UMC::YUY2:   par.info.color_format = UMC::Y210; break;
+            case UMC::YUV444: par.info.color_format = UMC::Y410; break;
+
+            default:
+                VM_ASSERT(!"Unknown subsampling");
+                return UMC::UMC_ERR_UNSUPPORTED;
+            }
+        }
+        else if (sh.color_config.BitDepth == 12)
+        {
+            switch (par.info.color_format)
+            {
+            case UMC::NV12:   par.info.color_format = UMC::P016; break;
+            case UMC::YUY2:   par.info.color_format = UMC::Y216; break;
+            case UMC::YUV444: par.info.color_format = UMC::Y416; break;
+
+            default:
+                VM_ASSERT(!"Unknown subsampling");
+                return UMC::UMC_ERR_UNSUPPORTED;
+            }
+        }
+
+        par.info.interlace_type = UMC::PROGRESSIVE;
+        par.info.aspect_ratio_width = par.info.aspect_ratio_height = 1;
+
+        par.lFlags = 0;
+
+        par.film_grain = sh.film_grain_param_present;
+
+        return UMC::UMC_OK;
+    }
+
+    void AV1Decoder::SetDPBSize(uint32_t size)
+    {
+        VM_ASSERT(size > 0);
+        VM_ASSERT(size < 128);
+
+        dpb.resize(size);
+        std::generate(std::begin(dpb), std::end(dpb),
+            [] { return new AV1DecoderFrame{}; }
+        );
+    }
+    void AV1Decoder::SetRefSize(uint32_t size)
+    {
+        VM_ASSERT(size > 0);
+        VM_ASSERT(size < 128);
+
+        refs_temp.resize(size);
+        std::generate(std::begin(refs_temp), std::end(refs_temp),
+           [] { return new AV1DecoderFrame{}; }
+        );
+    }
+
+    void AV1Decoder::CompleteDecodedFrames(FrameHeader const& fh, AV1DecoderFrame* pCurrFrame, AV1DecoderFrame* pPrevFrame)
+    {
+        if (pPrevFrame && Curr)
+        {
+            FrameHeader const& FH_OutTemp = Curr->GetFrameHeader();
+            if (Poutput->UID == -1)
+            {
+               Poutput = pPrevFrame;
+            }
+            else
+            {
+                if (Repeat_show || FH_OutTemp.show_frame)
+                {
+                    Poutput ->DecrementReference();
+                    Poutput = Curr;
+                }
+                else
+                {
+                    Curr->DecrementReference();
+                }
+            }
+        }
+
+        Curr = pCurrFrame;
```

Comment:
Maybe I missed something, but here simply we overwrite a pointer to `Curr` (that was initally allocated as `Curr(new AV1DecoderFrame{})` at ctor) w/ a frame from DPB (see `StartFrame`) - 
```
pCurrFrame = pFrameInProgress ?
   pFrameInProgress : StartFrame(fh, updated_refs, pPrevFrame); 

CompleteDecodedFrames(fh, pCurrFrame, pPrevFrame)
```
Please ensure we don't have a leak here. The same for `Poutput/Curr_temp`

Reply:
We may need add 
if (Curr)
    Curr->DecrementReference();
I will discuss this with @stasefremov and @IlyaSergeyev.

Reply:
We plan to fix it later.